fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/codec/TestingCodec.java

```diff
@@ -0,0 +1,197 @@
+    private static boolean disableLaziness;
+    private static boolean disableLazinessForLiveDocs;
+
+    public TestingCodec() {
+        super("RLT");
+        underlying = new LuceneOptimizedCodec(Lucene87Codec.Mode.BEST_SPEED);
+    }
+
+    /**
+     * Some of the tests expect to be able to continue reading from the associated readers after the underlying file
+     * has been deleted; this wraps the readers and forces them to read data when constructed.
+     * @param disableLaziness {@code true} to disable laziness, {@code false} to set it back to the default
+     */
+    public static void setDisableLaziness(boolean disableLaziness) {
+        TestingCodec.disableLaziness = disableLaziness;
+    }
+
+    /**
+     * Some of the tests expect to be able to continue reading from the associated readers after the underlying file
+     * has been deleted; this wraps the readers and forces the live docs to be read data when constructed.
+     * @param disableLazinessForLiveDocs {@code true} to disable laziness, {@code false} to set it back to the default
+     */
+    public static void setDisableLazinessForLiveDocs(final boolean disableLazinessForLiveDocs) {
+        TestingCodec.disableLazinessForLiveDocs = disableLazinessForLiveDocs;
+    }
+
+    @Override
+    public PostingsFormat postingsFormat() {
+        return underlying.postingsFormat();
+    }
+
+    @Override
+    public DocValuesFormat docValuesFormat() {
+        final DocValuesFormat underlyingDocValuesFormat = underlying.docValuesFormat();
+        if (disableLaziness) {
+            return new DocValuesFormat(underlyingDocValuesFormat.getName()) {
+                @Override
+                public DocValuesConsumer fieldsConsumer(final SegmentWriteState state) throws IOException {
+                    return underlyingDocValuesFormat.fieldsConsumer(state);
+                }
+
+                @Override
+                public DocValuesProducer fieldsProducer(final SegmentReadState state) throws IOException {
+                    final DocValuesProducer underlyingProducer = underlyingDocValuesFormat.fieldsProducer(state);
+                    underlyingProducer.ramBytesUsed();
+                    return underlyingProducer;
+                }
+            };
+        } else {
+            return underlyingDocValuesFormat;
+        }
+    }
+
+    @Override
+    public StoredFieldsFormat storedFieldsFormat() {
+        final StoredFieldsFormat storedFieldsFormat = underlying.storedFieldsFormat();
+        if (disableLaziness) {
+            return new StoredFieldsFormat() {
+                @Override
+                public StoredFieldsReader fieldsReader(final Directory directory, final SegmentInfo si, final FieldInfos fn, final IOContext context) throws IOException {
+                    final StoredFieldsReader storedFieldsReader = storedFieldsFormat.fieldsReader(directory, si, fn, context);
+                    storedFieldsReader.ramBytesUsed();
```

Comment:
This would only work for the "traditional" reader, right? The Optimized one will not be affected by `ramBytesUsed` call. On the other hand, maybe its lazy initialization is not problematic?